packages/airnode-deployer/src/handlers/aws/index.ts

```diff
@@ -104,6 +104,17 @@ export async function processHttpRequest(
     format: parsedConfig.nodeSettings.logFormat,
     level: parsedConfig.nodeSettings.logLevel,
   });
+
+  // Check if the request origin header is allowed in the config
+  const originVerification = verifyRequestOrigin(
```

Comment:
It's better to wrap the whole thing in:
```
if (event.httpMethod === 'OPTIONS')...
```

Reply:
hmm we need  the CORS headers returned for the `POST` method as well, right?

Reply:
Your implementation currently does set it only for OPTIONS requests. But I think setting the headers for OPTIONS is enough (https://stackoverflow.com/a/71429784).

Reply:
Hmm I do set them also for the POST method [here](https://github.com/api3dao/airnode/pull/1357/files#diff-6f3bd240ed3040ed5cbd29ed7ea1b83c7735fdcf318dddf814c7846f493bc830R137) (and they are included when testing with a react app)

Reply:
The [GCP docs](https://cloud.google.com/functions/docs/samples/functions-http-cors#code-sample) do add the Origin header and [AWS docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-cors.html) seem to add all

:thinking:

Reply:
Maybe it's a convention (or they simply did that out of convenience). My main point is not to call the gateway when you receive the OPTIONS request, but only return early the necessary headers. We can include the headers on the POST, I can't see an issue with that.

Reply:
> My main point is not to call the gateway when you receive the OPTIONS request, but only return early the necessary headers.

I'm afraid I don't quite follow what you mean by this. 

For AWS it is possible to setup a mock response for the gateway to handle the options method (so that the lambda is not called), but this is not possible for GCP so I thought it would be better to handle both cloud providers the same way.

Reply:
> I'm afraid I don't quite follow what you mean by this.

Sorry for the confusion, I realized I typed it wrong. I see that you return early when the OPTIONS request is successful, but we are missing an early return when the OPTIONS origin verification fails

Reply:
> but we are missing an early return when the OPTIONS origin verification fails

Ah I see what you mean, I'll add a status 400 and an error message here :+1: